addons/mail/static/src/components/discuss/tests/discuss_tests.js

```diff
@@ -4150,17 +4150,132 @@ QUnit.test('messages marked as read move to "History" mailbox', async function (
                 this.env.messaging.history.localId
             }"]
         `).classList.contains('o-active'),
-        "History mailbox should be active"
+        "history mailbox should be active"
     );
     assert.strictEqual(
         document.querySelectorAll(`.o_Discuss_thread .o_MessageList_empty`).length,
         0,
-        "History mailbox should not be empty after mark as read"
+        "history mailbox should not be empty after mark as read"
     );
     assert.strictEqual(
         document.querySelectorAll(`.o_Discuss_thread .o_MessageList_message`).length,
         2,
-        "History mailbox should have 2 messages"
+        "history mailbox should have 2 messages"
+    );
+});
+
+QUnit.test('single message marked as read should move to "History" mailbox and not others', async function (assert) {
+    assert.expect(9);
+
+    this.data['mail.message'].records = [{
+        author_id: [5, 'Demo User'],
+        body: '<p>test 1</p>',
+        id: 1,
+        needaction: true,
+        needaction_partner_ids: [3],
+    }, {
+        author_id: [6, 'Test User'],
+        body: '<p>test 2</p>',
+        id: 2,
+        needaction: true,
+        needaction_partner_ids: [3],
+    }];
+    this.data['mail.notification'].records = [{
+        id: 50,
+        is_read: false,
+        mail_message_id: 1,
+        res_partner_id: 3,
+    }, {
+        id: 51,
+        is_read: false,
+        mail_message_id: 2,
+        res_partner_id: 3,
+    }];
+    this.data.initMessaging.needaction_inbox_counter = 2;
+    await this.start({
+        discuss: {
+            params: {
+                default_active_id: 'mail.box_history',
+            },
+        },
+    });
```

Comment:
`3` is a magic value (corresponds to fallback current partner id). Would make that explicit in test by providing session to `env` with this value.